common/src/main/java/com/wynntils/commands/wynntils/WynntilsConfigCommand.java

```diff
@@ -332,4 +489,127 @@ private static int resetAllConfigOptions(CommandContext<CommandSourceStack> cont
+        String valueString = value == null ? "Value is null." : value.toString();
+
+        return new TextComponent("\n - ")
+                .withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent(configNameString)
+                        .withStyle(style -> style.withHoverEvent(new HoverEvent(
+                                HoverEvent.Action.SHOW_TEXT,
+                                new TextComponent("Description: " + config.getDescription())
+                                        .withStyle(ChatFormatting.LIGHT_PURPLE))))
+                        .withStyle(ChatFormatting.YELLOW)
+                        .append(new TextComponent(configTypeString).withStyle(ChatFormatting.WHITE))
+                        .append(new TextComponent(": "))
+                        .append(new TextComponent(valueString)
+                                .withStyle(ChatFormatting.GREEN)
+                                .withStyle(style -> style.withHoverEvent(new HoverEvent(
+                                        HoverEvent.Action.SHOW_TEXT,
+                                        new TextComponent("Click here to change this setting."))))));
+    }
+
+    private static MutableComponent getComponentForOverlay(Overlay overlay) {
+        return new TextComponent("\n - ")
+                .withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent(overlay.getShortName()).withStyle(ChatFormatting.AQUA));
+    }
+
+    private static MutableComponent getSpecificConfigComponent(ConfigHolder config) {
+        Object value = config.getValue();
+
+        String valueString = value == null ? "Value is null." : value.toString();
+        String configTypeString = "(" + config.getType().getSimpleName() + ")";
+
+        MutableComponent response = new TextComponent("");
+        response.append(new TextComponent("Config option: ")
+                .withStyle(ChatFormatting.WHITE)
+                .append(new TextComponent(config.getDisplayName()).withStyle(ChatFormatting.YELLOW))
+                .append("\n"));
+
+        response.append(new TextComponent("Value: ")
+                        .withStyle(ChatFormatting.WHITE)
+                        .append(new TextComponent(configTypeString))
+                        .append(new TextComponent(": "))
+                        .append(new TextComponent(valueString).withStyle(ChatFormatting.GREEN)))
+                .append("\n");
+        response.append(new TextComponent("Subcategory: ")
+                        .withStyle(ChatFormatting.WHITE)
+                        .append(new TextComponent(config.getMetadata().subcategory())))
+                .append("\n");
+        response.append(new TextComponent("Description: ")
+                        .withStyle(ChatFormatting.WHITE)
+                        .append(new TextComponent(config.getDescription())))
+                .append("\n");
+        return response;
+    }
```

Comment:
I think this looks good (barring the discussion of how overlay config names should be built, if they should belong to a feature or not).

But.

The entire class is very repetetive. At some point, we might consider seeing if we can raise the abstraction a bit, or if that is just hopeless given the Brigadier API...

Reply:
Yeah, I tried my best, but wasn't really my intention to recode this fully.